### DISCUSSION TITLE
Insert user to auth.users, get the latest id inserted and then add additional informations to public.profiles based on that last id?

### DISCUSSION THREAD

**Discussion by sadiki-o** [supabase/supabase#5287] [Feature Requests]
Hey guys i am trying to get this function to work, help please
note: if i have tried to create a user with signup  and then add additional informations to public.profiles but when the sigup function runs in front end it returns a user and whatever comes after it never get to execute that is why i thought about making a function that does all in one.
Function definition:

```
create or replace function insertUser( p text, n text, e text, s text, pass text)
returns void 
language sql
declare i uuid
as $$

  insert into auth.users( aud, role, email, encrypted_password, email_confirmed_at, created_at, raw_app_meta_data)
  values('authenticated', 'authenticated', e, crypt(pass, gen_salt('bf')), now(), now(), '{"provider":"email","providers":["email"]}');

  i := (select id from auth.users order by created_at limit 1);

  insert into public.profiles(i, nom, prenom, stam, email, created_at)
  values(i, n , p , s, e, now())
  
$$
```

**Comment by GaryAustin1** [comment]
Someone was trying to import users yesterday and it is more complicated than just inserting stuff into the user table...  appears other tables are involved.  Note they gave up and just curl for what they needed calling auth.  

https://github.com/supabase/supabase/discussions/5248

edit: also won't you be calling this function as a non-logged in user?  Would need security definer on the function and then the whole thing does not seem like a good idea...

**Reply by sadiki-o** [reply]
i have no idea know how can i add any additional information about users while supabase team say don't change auth.users ...
creating a second table public.profiles only help inserting the id of the new created user through a trigger ... in my case i have name, last name and any other columns that i want to fill at the creation of the user ...

**Reply by sadiki-o** [reply]
i think it is something that they might consider adding...

**Reply by GaryAustin1** [reply]
If you search around here you might find some other help on this as I've seen it discussed several times, but don't remember if it was solved.  Unless you are going to do things like above from a server call, it just seems like a security problem.  No matter what trying to fake out gotrue stuff and then have it work with gotrue long term is also a problem if they change columns in auth.

**Reply by sadiki-o** [reply]
also the signup function they provide in order to signup new users always kick the current user out which is not a good behavior in case someone wanted to create an application with an admin route where he can manipulate users and their data including creating users, deleting users, updating users and so on

**Reply by sadiki-o** [reply]
> If you search around here you might find some other help on this as I've seen it discussed several times, but don't remember if it was solved. Unless you are going to do things like above from a server call, it just seems like a security problem. No matter what trying to fake out gotrue stuff and then have it work with gotrue long term is also a problem if they change columns in auth.

yes ... i have searched but couldn't find anything ...

**Reply by GaryAustin1** [reply]
So doing a trigger function on auth.users insert and using meta_data provided in the signUp function won't work for you?

**Reply by sadiki-o** [reply]
> So doing a trigger function on auth.users insert and using meta_data provided in the signUp function won't work for you?

seems like an interresting idea, can u think of an implementation because i will have to sign up the user using supabase function and then insert meta data into the profiles table which is a bit challenging

**Reply by GaryAustin1** [reply]
https://supabase.com/docs/guides/auth/managing-user-data

Towards the end is a trigger function to get user data as the auth.user row is created.  It does not show getting the meta_data, but that works.

Here someone uses the metadata:
https://github.com/supabase/supabase/discussions/3787

**Reply by sadiki-o** [reply]
one last question how can i pass for example first_name, last_name as meta data to the signup function, i apologize for asking lot since i am not familiar with those new concepts

**Reply by GaryAustin1** [reply]
https://supabase.com/docs/reference/javascript/auth-signup#sign-up-with-additional-user-meta-data

**Reply by sadiki-o** [reply]
you are a life saver i can't thank you enough for the help u provided, THANK you very much
